packages/flutter/lib/src/material/tabs.dart

```diff
@@ -49,6 +49,40 @@ enum TabBarIndicatorSize {
   label,
 }
 
+/// Defines how tabs are aligned horizontally in a [TabBar].
+///
+/// See also:
+///
+///   * [TabBar], which displays a row of tabs.
+///   * [TabBarView], which displays a widget for the currently selected tab.
+///   * [TabBar.tabAlignment], which defines the horizontal alignment of the
+///     tabs within the [TabBar].
+enum TabAlignment {
+  /// When [TabBar.isScrollable] is true, tabs are aligned to the
+  /// start of the [TabBar] with a horizontal offset of 56.0 pixels.
+  ///
+  /// If [TabBar.isScrollable] is false, this is ignored.
+  ///
+  /// It is not recommended to set [TabAlignment.start] when
+  /// [ThemeData.useMaterial3] is false.
+  start,
+  /// When [TabBar.isScrollable] is true, tabs are aligned to the
+  /// start of the [TabBar].
+  ///
+  /// If [TabBar.isScrollable] is false, this is ignored.
+  ///
+  /// It is not recommended to set [TabAlignment.startOffset] when
+  /// [ThemeData.useMaterial3] is false.
+  startOffset,
+  /// When [TabBar.isScrollable] is false, tabs are stretched to fill the
+  /// [TabBar].
+  ///
+  /// If [TabBar.isScrollable] is true, this is ignored.
```

Comment:
Rather than a lot of ignored cases, can there be an assertion where it is set that iScrollable and alignment cannot both be set?

Reply:
I wanted to add assertions but I went for a soft approach.  

Android doesn't throw an error if you use unsupported alignment, it just ignores it but `NavigationRail` throws an error if you use unsupported alignment when `extended: true` for instance. 


I think adding a helpful message with an assertion would be a better idea to avoid confusion.